### PR TITLE
Move PreReceiveSecretDetectionEnabled to Project

### PR DESCRIPTION
**Pull request by gravis**
It was added to the wrong location with !1984

### CODE REVIEW

**Comment by gravis** [comment]
Sorry for that!

File: projects.go [hunk 1]
```diff
@@ -169,6 +169,7 @@ type Project struct {
 	MergeRequestDefaultTargetSelf            bool               `json:"mr_default_target_self"`
 	ModelExperimentsAccessLevel              AccessControlValue `json:"model_experiments_access_level"`
 	ModelRegistryAccessLevel                 AccessControlValue `json:"model_registry_access_level"`
+	PreReceiveSecretDetectionEnabled         *bool              `url:"pre_receive_secret_detection_enabled,omitempty" json:"pre_receive_secret_detection_enabled,omitempty"`
```
Comment (svanharmelen) [inline projects.go:172]:
```suggestion
	PreReceiveSecretDetectionEnabled         bool              `json:"pre_receive_secret_detection_enabled"`
```

**Review by svanharmelen — CHANGES_REQUESTED** [review]
Guess that is why I normally don't merge PRs with any undocumented fields 😏

**Comment by gravis** [comment]
> Guess this is why I normally don't merge PRs with undocumented features 😏

@svanharmelen The documentation for this feature is [here](https://docs.gitlab.com/ee/user/application_security/secret_detection/secret_push_protection/), but I think you are referring to https://docs.gitlab.com/ee/api/projects.html#get-single-project?

**Review by svanharmelen — APPROVED** [review]

**Comment by svanharmelen** [comment]
Yes indeed... But all is good. Thanks for fixing the issue 👍🏻

**Pull request merged**